module/src/clients/surface_temp.c

```diff
@@ -118,8 +135,19 @@ static int ssam_temp_hwmon_read_string(struct device *dev,
 	if (attr != hwmon_temp_label)
 		return -EOPNOTSUPP;
 
-	// TODO: get label from SAM
-	*str = "<not yet implemented>";
+	// Get the label from the SSAM if we haven't retrieved it yet.
+	if (ssam_temp->names[channel][0] == 0)
+	{
+		status =  ssam_retry(__ssam_tmp_get_name, ssam_temp->sdev->ctrl,
+				     SSAM_SSH_TID_SAM, channel + 1, &name_resp);
```

Comment:
Please use `sdev->uid.target` instead of  `SSAM_SSH_TID_SAM` here. If I remember correctly there might also be a second set of temperature sensors on the KIP target on some devices.